test/unit-test/mqtt_agent_utest.c

```diff
@@ -1305,6 +1354,19 @@ void test_MQTTAgent_CommandLoop_with_eventCallback( void )
     /* Ensure that callback is invoked. */
     TEST_ASSERT_EQUAL( 2, commandCompleteCallbackCount );
 
+    /* Invoking mqttEventCallback with MQTT_PACKET_TYPE_CONNECT packet type to cover default case. */
+    packetType = MQTT_PACKET_TYPE_CONNECT;
```

Comment:
This is literally not possible. If there's a path in the switch that can't be covered otherwise then that should be a code change.